metrics/core.go

```diff
@@ -184,6 +190,7 @@ type CacheMetrics interface {
 	RecordPutBadRequest()
 	RecordPutTotal()
 	RecordPutDuration(duration time.Duration)
+	RecordPutKeyProvided()
```

Comment:
Where does this get called by the app?

Reply:
Yes. This metric gets called from the `endpoints/put.go` in pull request #90